lib/govuk_index/migrated_formats.rb

```diff
@@ -39,5 +41,15 @@ def convert_to_allowed_hash(formats)
         end
       end
     end
+
+    def published_by_content_publisher?(format, app)
+      return false unless app == "content-publisher"
+      indexable_formats[format].include?("content-publisher")
+    end
+
+    def published_by_whitehall?(format, app)
+      return false unless app == "whitehall"
+      non_indexable_formats[format].include?("whitehall")
+    end
```

Comment:
Maybe we can make these methods a bit more generic to not need to have the publishing app hardcoded into them and instead take input from the yaml file?

Reply:
I've refactors this here https://github.com/alphagov/rummager/pull/1276/commits/784a28d31217fa8e607b7f2072b94a97d44290f9 - let me know what you think.